lib/Jsrt/Core/JsrtContextCore.cpp

```diff
@@ -120,6 +120,36 @@ HRESULT ChakraCoreHostScriptContext::FetchImportedModule(Js::ModuleRecordBase* r
     return E_INVALIDARG;
 }
 
+HRESULT ChakraCoreHostScriptContext::FetchImportedModuleFromScript(JsSourceContext dwReferencingSourceContext, LPCOLESTR specifier, Js::ModuleRecordBase** dependentModuleRecord)
+{
+    if (fetchImportedModuleFromScriptCallback == nullptr)
+    {
+        return E_INVALIDARG;
+    }
+
+    HRESULT hr = NOERROR;
+    Js::JavascriptString* specifierVar = nullptr;
+    specifierVar = Js::JavascriptString::NewCopySz(specifier, GetScriptContext());
+    
+    if (FAILED(hr))
+    {
+        return hr;
+    }
+
+    JsModuleRecord dependentRecord = JS_INVALID_REFERENCE;
```

Comment:
Refactored in latest iteration.

---
In reply to: [115328606](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r115328606) [](ancestors = 115328606)